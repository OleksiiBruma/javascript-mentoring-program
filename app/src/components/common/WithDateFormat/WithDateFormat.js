import { Component } from "react";
class WithDateFormat extends Component {
  convert = initialDate => {
    const date = new Date(initialDate);
    const dd = date.getDate() < 10  ? "0" + date.getDate(): date.getDate();
    const mm = date.getMonth()  < 10  ? `0${date.getMonth()+1}`  : date.getMonth() + 1 ;
    const yyyy = date.getFullYear();
    return `${dd}.${mm}.${yyyy}`;
  };
  render() {
    return this.props.render(this.convert);
  }
}

export default WithDateFormat;
