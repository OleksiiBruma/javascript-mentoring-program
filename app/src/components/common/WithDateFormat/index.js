import WithDateFormat from "./WithDateFormat"

export default WithDateFormat

