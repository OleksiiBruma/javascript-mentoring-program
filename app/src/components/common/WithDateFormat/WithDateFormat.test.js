import renderer from "react-test-renderer";
import React from "react";
import WithDateFormat from "../WithDateFormat/WithDateFormat";

it("return formated date", () => {
  const component = renderer.create(<WithDateFormat render={(convert)=>convert("12/02/2018")}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

