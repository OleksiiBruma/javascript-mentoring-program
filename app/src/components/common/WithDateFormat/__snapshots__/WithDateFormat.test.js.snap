// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return formated date 1`] = `"02.12.2018"`;
